### ISSUE TITLE
array.view with no arguments

### ISSUE THREAD

**Issue by zklaus** [dask/dask#5175]
Numpy arrays allow a call to `.view()` with no arguments passed. This constructs a new numpy array that is backed by the same data, ie the same memory as the original array and is useful when you need to just locally change the metadata, like the shape.

Dask requires the first argument, `dtype`, breaking some otherwise agnostic code.

In the MVE, the numpy part works, the dask part shows the breakage.
```
import dask.array as da
import numpy as np

n = np.array([1, 2, 3])
v = n.view()
v[1] = 10
print(n)

d = da.from_array(n)
dv = d.view()
dv[2] = 20
print(d)
```

Would you consider changing dask to also accept calls without arguments?

**Comment by TomAugspurger** [comment]
Yep, that seems reasonable.

**Comment by jrbourbeau** [comment]
@zklaus are you interested in submitting a PR? (no obligation though)

**Comment by zklaus** [comment]
Sorry, too swamped at the moment and on the brink of my summer vacation.

I did have a quick look, but there are too many things confusing me at the moment, both in the original numpy implementation and in the view handling of dask. For example it seems that
`dask_array[:]` is not giving a view, but the original array, in contrast to numpy?

If this is still open at the end of September, I might have another look.

**Comment by andersy005** [comment]
A fix for this issue went in as of #5736.  

```python
In [15]: import numpy as np

In [16]: import dask.array as da

In [17]: n = np.array([1, 2, 3])

In [18]: d = da.from_array(n)

In [19]: dv = d.view()
```

Is it ok to close this?

**Comment by jrbourbeau** [comment]
Thanks for following up on this issue @andersy005! Closing
